specs/eventing/data-plane.md

```diff
@@ -1,125 +1,171 @@
-
-A Sink MAY support `Batched Content Mode` but that mode is not used in Knative
-Eventing currently (that may change in future).
-
-### Retries
-
-Sinks should expect that retries and accept possibility that duplicate events
-may be delivered.
-
-### Error handling
-
-If Sink is not returning HTTP success header (200 or 202) then the event may be
-sent again. If the event can not be delivered then some sources of events (such
-as Knative sources, brokers or channels) MAY support
-[dead letter sink or channel](https://github.com/knative/eventing/blob/main/docs/delivery/README.md) for events that can not be
-delivered.
+[structured](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+content modes of the CloudEvetns HTTP binding. Senders MUST support both
+cleartext (`http`) and TLS (`https`) URLs as event delivery destinations.
+
+### HTTP Verbs
+
+In the absence of specific delivery preferences, the sender MUST initiate
+delivery of the event to the recipient using the HTTP POST verb, using either
+the structured or binary encoding of the event (sender's choice). This delivery
+SHOULD be performed using the CloudEvents HTTP Binding, version 1.
+
+Senders MAY probe the recipient with an [HTTP OPTIONS
+request](https://tools.ietf.org/html/rfc7231#section-4.3.7); if implemented, the
+recipent MUST indicate support for the POST verb using the [`Allow`
+header](https://tools.ietf.org/html/rfc7231#section-7.4.1). Senders which
+receive an error when probing with HTTP OPTIONS SHOULD proceed using the HTTP
+POST mechanism.
+
+### Event Acknowledgement and Repeat Delivery
+
+Event recipients MUST use the HTTP response code to indicate acceptance of an
+event. The recipient MUST NOT return a response accepting the event until it has
+handled event (processed the event or stored it in stable storage). The
+following response codes are explicitly defined; event recipients MAY also
+respond with other response codes. A response code not in this table SHOULD be
+treated as a retriable error.
+
+| Response code | Meaning                     | Retry | Delivery completed | Error |
+| ------------- | --------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)               | Yes\* | No\*               | Yes\* |
+| `200`         | [Event reply](#event-reply) | No    | Yes                | No    |
+| `202`         | Event accepted              | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)               | Yes\* | No\*               | Yes\* |
+| other `3xx`   | (Unspecified)               | Yes\* | No\*               | Yes\* |
+| `400`         | Unparsable event            | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist     | No    | No                 | Yes   |
```

Comment:
It seems like a 404 should be tried again, right? what if it comes back?

Reply:
https://github.com/knative/specs/blob/main/specs/eventing/data-plane.md#http-support only specifies:

> If an HTTP request's URL does not correspond to an existing endpoint, then the Sink MUST respond with 404 Not Found.

And doesn't say whether retries should be supported or not. I'm happy to change this.

Reply:
@n3wscott @evankanderson I think this: https://github.com/knative/eventing/blob/main/pkg/kncloudevents/message_sender.go#L192-L202
as a good section on 4xx errors

Reply:
I think I changed this to be largely in line with the link; 404, 409, and 429 are retried, and other 4xx errors are not, with 400 specifically being used to indicate a non-parseable event.

3xx, unspecified 2xx and 1xx are stell errors from an event-delivery point of view.